retrieval-market.md

```diff
@@ -21,62 +21,68 @@ The main components are as follows:
 
-```go
+```sh
 type RetDealProposal struct {
-	// Ref is the cid of the data to be retrieved
-	Ref Cid
+	## Reference to the data being retrieved.
+	ref Link
 
-	// Price is the total amount that the client is willing to pay for the
-	// retrieval of the data
-	Price TokenAmount
+	## The total amount that the client is willing to pay for the retrieval of the data.
+	price TokenAmount
 
-	// Payment is a payment info from the client to the retrieval miner for the data
-	Payment PaymentInfo
+	## The info from the client to the retrieval miner for the data
+	payment PaymentInfo
 }
 
-type ResponseStatus uint
-
-const (
-	Unset = ResponseStatus(iota)
-	Accepted
-	Rejected
-	Error
-)
+type RetDealResponse union {
+    | AcceptedResponse 0
+    | RejectedResponse 1
+    | ErrorResponse 2
+}
 
-type RetDealResponse struct {
-	Status  ResponseStatus
-	Message string
+type AcceptedResponse struct {}
+type RejectedResponse struct {
+    message String optional
 }
 
+type ErrorResponse RejectedResponse
+
 type Block struct {
-	// Prefix is the cid prefix parameters for this block. It describes how to
-	// hash the block to verify it matches the expected value.
-	Prefix CidPrefix
-	Data   []byte
+	## Cid prefix parameters for this block. It describes how to
+	## hash the block to verify it matches the expected value.
+	refix CidPrefix
+	data Bytes
+}
+
+## Represents all the metadata of a Cid.
+## It does not contains  any actual content information.
+type CidPrefix struct {
+	version  UInt
+	codec    UInt
+	mhType   UInt
+	mhLength UInt
 }
 ```
 
-`Retrieval miners` should also support a query service that allows clients to request pricing information from a miner. 
+`Retrieval miners` should also support a query service that allows clients to request pricing information from a miner.
 
-The query should include the CID of the piece that the client is interested in retrieving. The response contains whether or not the miner will serve that data, the price they will accept for it. 
+The query should include the CID of the piece that the client is interested in retrieving. The response contains whether or not the miner will serve that data, the price they will accept for it.
 
-```go
+```sh
 type RetQuery struct {
-	Piece Cid
+    ## TODO: what exactly does this link to?
+	piece Link
 }
 
-type RetQueryResponse struct {
-	Status   RetQueryStatus
-	MinPrice TokenAmount
+type RetQueryResponse union {
+    | AvailableResponse
+    | UnavailableResponse
```

Comment:
A `representation` clause on unions is required.  (See "Unions are a special case..." in [schema/Representations.md](https://github.com/ipld/specs/blob/afe04f38c8870044602e52eb14356f8a60c61d90/schema-layer/schemas/representations.md#kinds-and-their-representations).)

Reply:
okkkkk

Reply:
fixed